modules/perception/cuda_util/block_uf.cu

```diff
@@ -0,0 +1,218 @@
+/******************************************************************************
+ * Copyright (c) 2014, Victor Matheus de Araujo Oliveira All rights reserved.
```

Comment:
Are we able to use this license?

Reply:
The block-wise union find algorithm in CUDA-based implementation of connected component generator is modified from an open-source code which use this BSD license. We consulted with laywer and confirmed that we could use the related code given that "Redistributions in source or binary form must contain the copyright notice for the licensed software and the conditions and disclaimer contained in the license".